### ISSUE TITLE
visit dependency issue

### ISSUE THREAD

**Issue by rajeeja** [spack/spack#13555]
```console
$ spack install visit
==> Error: An unsatisfiable variant constraint has been detected for spec:

    vtk@8.1.2%gcc@8.3.0 build_type=RelWithDebInfo ~ffmpeg~mpi+opengl2~osmesa~python~qt~xdmf arch=linux-ubuntu19.04-sandybridge

while trying to concretize the partial spec:

    visit@3.0.1%gcc@8.3.0~adios2 build_type=RelWithDebInfo +gui+hdf5+mpi+python+silo arch=linux-ubuntu19.04-sandybridge
        ^cmake@3.15.4%gcc@8.3.0~doc+ncurses+openssl+ownlibs~qt arch=linux-ubuntu19.04-sandybridge
            ^ncurses
                ^pkgconf@1.6.3%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
            ^openssl
                ^perl@5.14.0:
                    ^gdbm
                        ^readline
                ^zlib@1.2.11%gcc@8.3.0+optimize+pic+shared arch=linux-ubuntu19.04-sandybridge

visit requires vtk variant +python, but spec asked for ~python
```

**Comment by adamjstewart** [comment]
This is a known bug in the concretizer that @tgamblin is currently working on fixing. In the meantime, the following workaround should work:
```console
$ spack install visit ^vtk+python+qt ^python@:2.8
```

**Comment by rajeeja** [comment]
Thanks, but now the error is:

spack install visit ^vtk+python+qt ^python@:2.8
==> Error: An unsatisfiable variant constraint has been detected for spec:

    hdf5@1.10.5%gcc@8.3.0~cxx~debug~fortran~hl+mpi+pic+shared~szip~threadsafe arch=linux-ubuntu19.04-sandybridge
        ^mpi
        ^zlib@1.2.11%gcc@8.3.0+optimize+pic+shared arch=linux-ubuntu19.04-sandybridge


while trying to concretize the partial spec:

    netcdf@4.7.0%gcc@8.3.0~dap~hdf4 maxdims=1024 maxvars=8192 +mpi~parallel-netcdf+pic+shared arch=linux-ubuntu19.04-sandybridge
        ^autoconf
            ^m4@1.4.18%gcc@8.3.0+sigsegv arch=linux-ubuntu19.04-sandybridge
                ^libsigsegv
            ^perl@5.30.0%gcc@8.3.0+cpanm+shared+threads arch=linux-ubuntu19.04-sandybridge
                ^gdbm@1.18.1%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
                    ^readline@8.0%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
                        ^ncurses@6.1%gcc@8.3.0~symlinks~termlib arch=linux-ubuntu19.04-sandybridge
                            ^pkgconf@1.6.3%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
        ^automake
        ^libtool
        ^mpi
        ^zlib@1.2.11%gcc@8.3.0+optimize+pic+shared arch=linux-ubuntu19.04-sandybridge


netcdf requires hdf5 variant +hl, but spec asked for ~hl

**Comment by adamjstewart** [comment]
Just keep tagging on the constraints you need:
```console
$ spack install visit ^vtk+python+qt ^python@:2.8 ^hdf5+hl
```
You can use `spack spec` to test things without doing the full install.

**Comment by rajeeja** [comment]
[+] /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/help2man-1.47.11-vbzxfbtlf2joc3bucjyqhan5hlek774q
==> Installing bison
==> Searching for binary cache of bison
==> No binary for bison found: installing from source
==> Fetching https://ftpmirror.gnu.org/bison/bison-3.4.2.tar.gz
################################################################################################################################################################## 100.0%
==> Staging archive: /tmp/jain/spack-stage/spack-stage-bison-3.4.2-t6otz3jwvvsvs4g46q273alqbq5wqhs6/bison-3.4.2.tar.gz
==> Created stage in /tmp/jain/spack-stage/spack-stage-bison-3.4.2-t6otz3jwvvsvs4g46q273alqbq5wqhs6
==> No patches needed for bison
==> Building bison [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'

1 error found in build log:
     492      GEN      lib/wctype.h
     493    make[1]: Entering directory '/tmp/jain/spack-stage/spack-stage-bison-3.4.2-t6otz3jwvvsvs4g46q273alqbq5wqhs6/spack-src/spack-build'
     494    make[1]: Entering directory '/tmp/jain/spack-stage/spack-stage-bison-3.4.2-t6otz3jwvvsvs4g46q273alqbq5wqhs6/spack-src/spack-build'
     495      LEX      examples/c/reccalc/scan.stamp
     496      LEX      examples/c/reccalc/scan.stamp
     497    make[1]: Leaving directory '/tmp/jain/spack-stage/spack-stage-bison-3.4.2-t6otz3jwvvsvs4g46q273alqbq5wqhs6/spack-src/spack-build'
  >> 498    mv: cannot stat 'examples/c/reccalc/scan.stamp.tmp': No such file or directory
     499    make[1]: *** [Makefile:9327: examples/c/reccalc/scan.stamp] Error 1
     500    make[1]: Leaving directory '/tmp/jain/spack-stage/spack-stage-bison-3.4.2-t6otz3jwvvsvs4g46q273alqbq5wqhs6/spack-src/spack-build'
     501    make: *** [Makefile:9319: examples/c/reccalc/scan.c] Error 2

See build log for details:
  /tmp/jain/spack-stage/spack-stage-bison-3.4.2-t6otz3jwvvsvs4g46q273alqbq5wqhs6/spack-build-out.txt

jain@jain:~/my_f1/source/Grid/localAPI$

**Comment by adamjstewart** [comment]
@rajeeja there was a known bug with building bison 3.4.2 in parallel that was fixed in #13231 and #13402. You can either update to the last commit on develop, cherry pick those particular PRs, or run `spack install -j1 bison` to build in serial.

**Comment by rajeeja** [comment]
cool thanks. Now we're at qt:
==> Installing qt
==> Searching for binary cache of qt
==> No binary for qt found: installing from source
==> Error: AttributeError: Query of package 'opengl' for 'libs' failed
	prefix : /opt/opengl
	spec : opengl@4.5.0%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
	queried as : opengl
	extra parameters : []

/home/jain/spack/lib/spack/spack/build_environment.py:792, in child_process:
        789            tb_string = traceback.format_exc()
        790
        791            # build up some context from the offending package so we can
  >>    792            # show that, too.
        793            package_context = get_package_context(tb)
        794
        795            build_log = None

**Comment by adamjstewart** [comment]
Based on that error message, I'm guessing you have `opengl` set in your `packages.yaml` as an external package, located in directory `/opt/opengl`. Can you share `ls /opt/opengl/lib*/*`? Spack checks in both `lib64` and `lib` for a `libGL.so` file.

**Comment by rajeeja** [comment]
ls /opt/opengl/lib*/*
ls: cannot access '/opt/opengl/lib*/*': No such file or directory

I recently upgraded to ubuntu19.04
Haven't changed packages.yaml


cat ~/.spack/packages.yaml 
packages:
  mpich:
    paths:
      mpich@3.2: /home/jain/install/mpich-3.2/
      openblas@0.2.19: /usr
    buildable: False
  openblas:
    paths:
      openblas@0.2.19: /usr
    buildable: False
  python:
    paths:
      python@2.7.12: /usr
    buildable: False
  jdk:
    paths:
      jdk@10.0.2 arch=linux-ubuntu18.04-x86_64: /usr
      buildable: False
  opengl:
    paths:
      opengl@4.5.0: /opt/opengl
      buildable: False

After removing 
opengl: ..
..
..

from the above

It is trying to build llvm and it is talking forever.

**Comment by adamjstewart** [comment]
Yeah, LLVM takes quite a while to build.

**Comment by rajeeja** [comment]
==> Installing qt
==> Searching for binary cache of qt
==> No binary for qt found: installing from source
==> Fetching http://download.qt.io/archive/qt/5.13/5.13.1/single/qt-everywhere-src-5.13.1.tar.xz
################################################################################################# 100.0%################################################################################################# 100.0%
==> Staging archive: /tmp/jain/spack-stage/spack-stage-qt-5.13.1-tfkismle4ybf5fhhvkqm7ph25r5oolbi/qt-everywhere-src-5.13.1.tar.xz
==> Created stage in /tmp/jain/spack-stage/spack-stage-qt-5.13.1-tfkismle4ybf5fhhvkqm7ph25r5oolbi
==> Applied patch /home/jain/spack/var/spack/repos/builtin/packages/qt/btn_trigger_happy.patch
==> Applied patch /home/jain/spack/var/spack/repos/builtin/packages/qt/qt5-pcre.patch
==> Ran patch() for qt
==> Building qt [Package]
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'

**Comment by adamjstewart** [comment]
There's no more to the error message? There should be a `spack-build-out.txt` log file containing a lot of useful info as to what crashed.

**Comment by rajeeja** [comment]
c++ -c -pipe -O3 -std=c++1z -fvisibility=hidden -fvisibility-inlines-hidden -Wall -W -D_REENTRANT -fPIC -DQT_NO_USING_NAMESPACE -DQT_NO_FOREACH -DELF_INTERPRETER=\"/lib64/ld-linux-x86-64.so.2\" -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_BUILD_CORE_LIB -DQT_BUILDING_QT -DQT_NO_CAST_TO_ASCII -DQT_ASCII_CAST_WARNINGS -DQT_MOC_COMPAT -DQT_USE_QSTRINGBUILDER -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050000 -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -DQT_NO_DEBUG -I. -Iglobal -I../3rdparty/harfbuzz/src -I../3rdparty/md5 -I../3rdparty/md4 -I../3rdparty/sha3 -I../3rdparty/forkfd -I../3rdparty/tinycbor/src -I../../include -I../../include/QtCore -I../../include/QtCore/5.13.1 -I../../include/QtCore/5.13.1/QtCore -I.moc -I.tracegen -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/pcre2-10.31-fsjdwptmxav3psx5k7gneycykl7ypsg7/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/glib-2.56.3-b2625ug5kzckcbaklcigv7vqzxdu4jqg/include/glib-2.0 -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/glib-2.56.3-b2625ug5kzckcbaklcigv7vqzxdu4jqg/lib/glib-2.0/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/pcre-8.42-xhbx6kzfxt47oskarxws5j6rr7mgqkrt/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/openssl-1.1.1d-b5y66l5lquboongswoujvst6rg5k5lne/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/freetype-2.10.1-hfjsj2i4zyexdkmog2egr4juohvtcafj/include/freetype2 -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/harfbuzz-2.3.1-yoiyfftgzotfzeo5pqt26gn2i6prpqrj/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/pcre2-10.31-fsjdwptmxav3psx5k7gneycykl7ypsg7/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/double-conversion-2.0.1-jzeaxvflvipyvncsv53thxwuhoohtudx/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libpng-1.6.37-wcmsoznvuqqcn7wuuhmakpk2edjbh7ei/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libjpeg-turbo-2.0.3-d2pcpxikmme6fsxyo75apouvrg2blh7b/include -I/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/zlib-1.2.11-qceg7yptdvihlbtdndt3ojdmc2lrovnr/include -I../../mkspecs/linux-g++ -o .obj/qatomic.o thread/qatomic.cpp
thread/qatomic.cpp:1624:4: error: #error "Q_ATOMIC_INT64_IS_SUPPORTED must be defined on a 64-bit platform"
 #  error "Q_ATOMIC_INT64_IS_SUPPORTED must be defined on a 64-bit platform"
    ^~~~~
In file included from ../../include/QtCore/qglobal.h:1,
                 from thread/qatomic.h:41,
                 from thread/qatomic.cpp:41:
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h: In instantiation of 'class QBasicAtomicInteger<long int>':
../../include/QtCore/../../src/corelib/thread/qatomic.h:55:7:   required from 'class QAtomicInteger<long int>'
thread/qatomic.cpp:1631:1:   required from here
../../include/QtCore/../../src/corelib/global/qglobal.h:121:63: error: static assertion failed: template parameter is an integral of a size not supported on this platform
 #  define Q_STATIC_ASSERT_X(Condition, Message) static_assert(bool(Condition), Message)
                                                               ^~~~~~~~~~~~~~~
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h:97:5: note: in expansion of macro 'Q_STATIC_ASSERT_X'
     Q_STATIC_ASSERT_X(QAtomicOpsSupport<sizeof(T)>::IsSupported, "template parameter is an integral of a size not supported on this platform");
     ^~~~~~~~~~~~~~~~~
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h: In instantiation of 'class QBasicAtomicInteger<long unsigned int>':
../../include/QtCore/../../src/corelib/thread/qatomic.h:55:7:   required from 'class QAtomicInteger<long unsigned int>'
thread/qatomic.cpp:1632:1:   required from here
../../include/QtCore/../../src/corelib/global/qglobal.h:121:63: error: static assertion failed: template parameter is an integral of a size not supported on this platform
 #  define Q_STATIC_ASSERT_X(Condition, Message) static_assert(bool(Condition), Message)
                                                               ^~~~~~~~~~~~~~~
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h:97:5: note: in expansion of macro 'Q_STATIC_ASSERT_X'
     Q_STATIC_ASSERT_X(QAtomicOpsSupport<sizeof(T)>::IsSupported, "template parameter is an integral of a size not supported on this platform");
     ^~~~~~~~~~~~~~~~~
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h: In instantiation of 'class QBasicAtomicInteger<long long unsigned int>':
../../include/QtCore/../../src/corelib/thread/qatomic.h:55:7:   required from 'class QAtomicInteger<long long unsigned int>'
thread/qatomic.cpp:1633:1:   required from here
../../include/QtCore/../../src/corelib/global/qglobal.h:121:63: error: static assertion failed: template parameter is an integral of a size not supported on this platform
 #  define Q_STATIC_ASSERT_X(Condition, Message) static_assert(bool(Condition), Message)
                                                               ^~~~~~~~~~~~~~~
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h:97:5: note: in expansion of macro 'Q_STATIC_ASSERT_X'
     Q_STATIC_ASSERT_X(QAtomicOpsSupport<sizeof(T)>::IsSupported, "template parameter is an integral of a size not supported on this platform");
     ^~~~~~~~~~~~~~~~~
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h: In instantiation of 'class QBasicAtomicInteger<long long int>':
../../include/QtCore/../../src/corelib/thread/qatomic.h:55:7:   required from 'class QAtomicInteger<long long int>'
thread/qatomic.cpp:1634:1:   required from here
../../include/QtCore/../../src/corelib/global/qglobal.h:121:63: error: static assertion failed: template parameter is an integral of a size not supported on this platform
 #  define Q_STATIC_ASSERT_X(Condition, Message) static_assert(bool(Condition), Message)
                                                               ^~~~~~~~~~~~~~~
../../include/QtCore/../../src/corelib/thread/qbasicatomic.h:97:5: note: in expansion of macro 'Q_STATIC_ASSERT_X'
     Q_STATIC_ASSERT_X(QAtomicOpsSupport<sizeof(T)>::IsSupported, "template parameter is an integral of a size not supported on this platform");
     ^~~~~~~~~~~~~~~~~
make[3]: *** [Makefile:13503: .obj/qatomic.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/tmp/jain/spack-stage/spack-stage-qt-5.13.1-tfkismle4ybf5fhhvkqm7ph25r5oolbi/spack-src/qtbase/src/corelib'
make[2]: *** [Makefile:169: sub-corelib-make_first] Error 2
make[2]: Leaving directory '/tmp/jain/spack-stage/spack-stage-qt-5.13.1-tfkismle4ybf5fhhvkqm7ph25r5oolbi/spack-src/qtbase/src'
make[1]: *** [Makefile:51: sub-src-make_first] Error 2
make[1]: Leaving directory '/tmp/jain/spack-stage/spack-stage-qt-5.13.1-tfkismle4ybf5fhhvkqm7ph25r5oolbi/spack-src/qtbase'
make: *** [Makefile:84: module-qtbase-make_first] Error 2

**Comment by adamjstewart** [comment]
@michaelkuhn @davydden @sethrj @Sinan81 have helped me a lot recently with tracking down Qt build failures. Maybe one of them has seen this before and knows what to do?

**Comment by michaelkuhn** [comment]
This should be fixed by #13262.

**Comment by sethrj** [comment]
Oh qt 🤮

**Comment by rajeeja** [comment]
ok. updated spack now this:

`-- Checking for VTK in /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/vtk-8.1.2-jjxziuq2btqd2evww2z33zlwmzgsz2rn/lib/cmake/vtk-8.1
--   VTK_FOUND=1
--   VTK_MAJOR_VERSION=8
--   VTK_MINOR_VERSION=1
--   VTK_BUILD_VERSION=2
--   VTK_INCLUDE_DIRS=/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/vtk-8.1.2-jjxziuq2btqd2evww2z33zlwmzgsz2rn/include/vtk-8.1;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/zlib-1.2.11-qceg7yptdvihlbtdndt3ojdmc2lrovnr/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/freetype-2.10.1-hfjsj2i4zyexdkmog2egr4juohvtcafj/include/freetype2;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/lz4-1.9.2-errz5ctxrloonhgx5ycxfodgig4tahpv/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/expat-2.2.9-gas6dsyuuuryoxt76yy2jyxnflzo3oa7/include;/usr/include/python2.7;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libjpeg-turbo-2.0.3-d2pcpxikmme6fsxyo75apouvrg2blh7b/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libpng-1.6.37-wcmsoznvuqqcn7wuuhmakpk2edjbh7ei/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libtiff-4.0.10-t7yyp3neemreni6gf5kqsohsvvur6kon/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-jfhrfhhwtdwbqlww2sp4437v4sp3muux/include;/home/jain/install/mpich-3.2/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/netcdf-c-4.7.2-lzvsmgcsjrikdxeflqde77nj3qxwraej/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libxml2-2.9.9-axrcnwtjkgm6sv4pvodrfprgalbvemip/include/libxml2;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libiconv-1.16-2zirdxdzr26y2merf7cv6sugwxlbkaud/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/netcdf-cxx-4.2-g3ialb6diicqeipkh3ossz7ysfaieqkq/include;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/jsoncpp-1.9.1-rbg6fii2arhxlkihonk4u6mhlvkg3nwj/include
--   VTK_DEFINITIONS=vtkDomainsChemistry_AUTOINIT=1(vtkDomainsChemistryOpenGL2);vtkIOExport_AUTOINIT=1(vtkIOExportOpenGL2);vtkRenderingContext2D_AUTOINIT=1(vtkRenderingContextOpenGL2);vtkRenderingCore_AUTOINIT=3(vtkInteractionStyle,vtkRenderingFreeType,vtkRenderingOpenGL2);vtkRenderingOpenGL2_AUTOINIT=1(vtkRenderingGL2PSOpenGL2);vtkRenderingVolume_AUTOINIT=1(vtkRenderingVolumeOpenGL2)
--   VTK_LIBRARIES=vtksys;vtkCommonCore;vtkCommonMath;vtkCommonMisc;vtkCommonSystem;vtkCommonTransforms;vtkCommonDataModel;vtkCommonColor;vtkCommonExecutionModel;vtkCommonComputationalGeometry;vtkFiltersCore;vtkFiltersGeneral;vtkImagingCore;vtkImagingFourier;vtkalglib;vtkFiltersStatistics;vtkFiltersExtraction;vtkInfovisCore;vtkFiltersGeometry;vtkFiltersSources;vtkRenderingCore;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/zlib-1.2.11-qceg7yptdvihlbtdndt3ojdmc2lrovnr/lib/libz.so;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/freetype-2.10.1-hfjsj2i4zyexdkmog2egr4juohvtcafj/lib/libfreetype.so;vtkRenderingFreeType;vtkRenderingContext2D;vtkChartsCore;vtkDICOMParser;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/lz4-1.9.2-errz5ctxrloonhgx5ycxfodgig4tahpv/lib/liblz4.so;vtkIOCore;vtkIOLegacy;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/expat-2.2.9-gas6dsyuuuryoxt76yy2jyxnflzo3oa7/lib/libexpat.so;vtkIOXMLParser;vtkDomainsChemistry;vtkglew;vtkRenderingOpenGL2;vtkDomainsChemistryOpenGL2;vtkIOXML;vtkParallelCore;vtkFiltersAMR;vtkFiltersFlowPaths;vtkFiltersGeneric;vtkImagingSources;vtkFiltersHybrid;vtkFiltersHyperTree;vtkImagingGeneral;vtkFiltersImaging;vtkFiltersModeling;vtkFiltersParallel;vtkFiltersParallelImaging;vtkFiltersPoints;vtkFiltersProgrammable;/usr/lib/x86_64-linux-gnu/libpython2.7.so;vtkWrappingTools;vtkWrappingPythonCore;vtkFiltersPython;vtkFiltersSMP;vtkFiltersSelection;vtkFiltersTexture;vtkFiltersTopology;verdict;vtkFiltersVerdict;vtkInteractionStyle;vtkGUISupportQt;vtkGUISupportQtOpenGL;vtkmetaio;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libjpeg-turbo-2.0.3-d2pcpxikmme6fsxyo75apouvrg2blh7b/lib/libjpeg.so;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libpng-1.6.37-wcmsoznvuqqcn7wuuhmakpk2edjbh7ei/lib/libpng.so;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libtiff-4.0.10-t7yyp3neemreni6gf5kqsohsvvur6kon/lib/libtiff.so;vtkIOImage;vtkImagingHybrid;vtkInfovisLayout;vtkImagingColor;vtkRenderingAnnotation;vtkRenderingVolume;vtkInteractionWidgets;vtkViewsCore;vtkproj4;vtkGeovisCore;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-jfhrfhhwtdwbqlww2sp4437v4sp3muux/lib/libhdf5.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so;/home/jain/install/mpich-3.2/lib/libmpi.so;/home/jain/install/mpich-3.2/lib/libmpicxx.so;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-jfhrfhhwtdwbqlww2sp4437v4sp3muux/lib/libhdf5_hl.so;vtkIOAMR;vtkIOEnSight;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/netcdf-c-4.7.2-lzvsmgcsjrikdxeflqde77nj3qxwraej/lib/libnetcdf.so;vtkexoIIc;vtkIOExodus;vtkgl2ps;vtkRenderingGL2PSOpenGL2;vtklibharu;vtkIOExport;vtkIOExportOpenGL2;vtkIOGeometry;vtkIOImport;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/libxml2-2.9.9-axrcnwtjkgm6sv4pvodrfprgalbvemip/lib/libxml2.so;vtkIOInfovis;vtkIOLSDyna;vtkIOMINC;vtkoggtheora;vtkIOMovie;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/netcdf-cxx-4.2-g3ialb6diicqeipkh3ossz7ysfaieqkq/lib/libnetcdf_c++.so;vtkIONetCDF;vtkIOPLY;/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/jsoncpp-1.9.1-rbg6fii2arhxlkihonk4u6mhlvkg3nwj/lib/libjsoncpp.so;vtkIOParallel;vtkIOParallelXML;vtksqlite;vtkIOSQL;vtkIOTecplotTable;vtkIOVideo;vtkImagingMath;vtkImagingMorphological;vtkImagingStatistics;vtkImagingStencil;vtkInteractionImage;vtkRenderingContextOpenGL2;vtkRenderingImage;vtkRenderingLOD;vtkRenderingLabel;vtkRenderingVolumeOpenGL2;vtkViewsContext2D;vtkViewsInfovis
--   VTK_INSTALL_PREFIX=/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/vtk-8.1.2-jjxziuq2btqd2evww2z33zlwmzgsz2rn
--   VTK_LIBRARY_DIRS=/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/vtk-8.1.2-jjxziuq2btqd2evww2z33zlwmzgsz2rn/lib
--   VTK_PY_WRAPPERS_DIR=/home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/vtk-8.1.2-jjxziuq2btqd2evww2z33zlwmzgsz2rn/lib/python2.7
-- Found VTK Python Wrappers - /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/vtk-8.1.2-jjxziuq2btqd2evww2z33zlwmzgsz2rn/lib/python2.7
-- Threading for VisIt: OFF
-- OSPRay for VisIt: OFF
-- Python scripting disabled because Python was not found.
-- Python Engine Filters disabled because Python was not found.
-- moc location: /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/qt-5.13.1-n3j62fmfeqlazc5dlmvi7fxd5v2dvxxh/bin/moc
-- Looking for QWT
--   Found library qwt in /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/qwt-6.1.3-f5fmxdp27ui7p7lua4iay7o3yxe2scam/lib
--   QWT found
-- Looking for ICET
--   ICET not requested
-- Looking for TCMALLOC
--   TCMALLOC not requested
-- Looking for ZLIB
--   Found library z in /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/zlib-1.2.11-qceg7yptdvihlbtdndt3ojdmc2lrovnr/lib
--   ZLIB found
-- Looking for HDF5
--   Found library hdf5 in /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-jfhrfhhwtdwbqlww2sp4437v4sp3muux/lib
--   HDF5 found
-- Looking for HDF5_MPI
CMake Error at CMake/SetUpThirdParty.cmake:214 (MESSAGE):
  Library hdf5_mpi not found in
  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-jfhrfhhwtdwbqlww2sp4437v4sp3muux/lib
Call Stack (most recent call first):
  CMake/FindHDF5.cmake:76 (SET_UP_THIRD_PARTY)
  CMakeLists.txt:1289 (INCLUDE)


-- Configuring incomplete, errors occurred!
See also "/tmp/jain/spack-stage/spack-stage-visit-3.0.1-hwvacjcphvmg3araiwzmq3ayagnlihgq/spack-build/CMakeFiles/CMakeOutput.log".
See also "/tmp/jain/spack-stage/spack-stage-visit-3.0.1-hwvacjcphvmg3araiwzmq3ayagnlihgq/spack-build/CMakeFiles/CMakeError.log".
`

**Comment by rajeeja** [comment]
=> spack spec visit ^vtk+python+qt  ^hdf5+hl
Input spec
--------------------------------
visit
    ^hdf5+hl
    ^vtk+python+qt

Concretized
--------------------------------
visit@3.0.1%gcc@8.3.0~adios2 build_type=RelWithDebInfo +gui+hdf5+mpi patches=b1c10e8f99fd869cc488e7c17e135849c7f0d4ca4414d18a6760247188114031 +python+silo arch=linux-ubuntu19.04-sandybridge
    ^cmake@3.15.4%gcc@8.3.0~doc+ncurses+openssl+ownlibs~qt arch=linux-ubuntu19.04-sandybridge
        ^ncurses@6.1%gcc@8.3.0~symlinks~termlib arch=linux-ubuntu19.04-sandybridge
            ^pkgconf@1.6.3%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
        ^openssl@1.1.1d%gcc@8.3.0+systemcerts arch=linux-ubuntu19.04-sandybridge
            ^perl@5.30.0%gcc@8.3.0+cpanm+shared+threads arch=linux-ubuntu19.04-sandybridge
                ^gdbm@1.18.1%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
                    ^readline@8.0%gcc@8.3.0 arch=linux-ubuntu19.04-sandybridge
            ^zlib@1.2.11%gcc@8.3.0+optimize+pic+shared arch=linux-ubuntu19.04-sandybridge
    ^hdf5@1.10.5%gcc@8.3.0~cxx~debug~fortran+hl+mpi patches=b61e2f058964ad85be6ee5ecea10080bf79e73f83ff88d1fa4b602d00209da9c +pic+shared~szip~threadsafe arch=linux-ubuntu19.04-sandybridge



and 

jain@jain:~/DoxygenFortranEx$ 
==> spack find -p hdf5
==> 7 installed packages
-- linux-ubuntu18.04-x86_64 / gcc@7.3.0 -------------------------
hdf5@1.8.19  /home/jain/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/hdf5-1.8.19-lhsjuvq3o3z46itveg72rs6z4vgb7bz5
hdf5@1.10.1  /home/jain/install/hdf5
hdf5@1.10.3  /home/jain/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/hdf5-1.10.3-4fdkqsobpzojitxaa7irv3kqwuv3nqb4
hdf5@1.10.3  /home/jain/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/hdf5-1.10.3-oajjpwx3wgvc2owmx5bbwmre5u36uwb2

-- linux-ubuntu19.04-sandybridge / gcc@8.3.0 --------------------
hdf5@1.8.19  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.8.19-pqljigsjs6extzhb67uodq5m5uafgizi
hdf5@1.8.19  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.8.19-klluprhhon4wwakzjsublza5pgazsuq5
hdf5@1.10.5  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-2d3rmyxlcmdaxmgpawmkf3fjhoz34ovg

**Comment by adamjstewart** [comment]
I'm confused why `spack spec` shows `hd5+mpi`, but doesn't show any MPI libraries. Is that the full output? Are you using an external package for your MPI?

According to `spack blame visit`, @sethrj @dorier @goxberry have contributed to the `visit` package in the past. They might be able to help.

**Comment by rajeeja** [comment]
yes I have external mpi:
packages:
  mpich:
    paths:
      mpich@3.2: /home/jain/install/mpich-3.2/
      openblas@0.2.19: /usr
    buildable: False

**Comment by adamjstewart** [comment]
Hmm, that doesn't look right. It should be:
```yaml
packages:
  all:
    providers:
      mpi:
      - mpich
      blas:
      - openblas
      lapack:
      - openblas
  mpich:
    paths:
      mpich@3.2: /home/jain/install/mpich-3.2/
    buildable: False
  openblas:
    paths:
      openblas@0.2.19: /usr
    buildable: False
```

**Comment by rajeeja** [comment]
Modified packages.yaml as you pointed out, it still gives the same error. I now have so many variants of hdf5 and it seems visit wants something else:)
cmake visit error:
``` 
    183    -- Looking for HDF5_MPI
  >> 184    CMake Error at CMake/SetUpThirdParty.cmake:214 (MESSAGE):
     185      Library hdf5_mpi not found in
     186      /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-jfhr
```
Many HDF5s!:
```
==> spack find -p hdf5
==> 8 installed packages
-- linux-ubuntu18.04-x86_64 / gcc@7.3.0 -------------------------
hdf5@1.8.19  /home/jain/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/hdf5-1.8.19-lhsjuvq3o3z46itveg72rs6z4vgb7bz5
hdf5@1.10.1  /home/jain/install/hdf5
hdf5@1.10.3  /home/jain/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/hdf5-1.10.3-4fdkqsobpzojitxaa7irv3kqwuv3nqb4
hdf5@1.10.3  /home/jain/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/hdf5-1.10.3-oajjpwx3wgvc2owmx5bbwmre5u36uwb2

-- linux-ubuntu19.04-sandybridge / gcc@8.3.0 --------------------
hdf5@1.8.19  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.8.19-pqljigsjs6extzhb67uodq5m5uafgizi
hdf5@1.8.19  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.8.19-klluprhhon4wwakzjsublza5pgazsuq5
hdf5@1.10.5  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-fk7bgqx7qml4exatuww6tjwzst34x5qb
hdf5@1.10.5  /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/hdf5-1.10.5-2d3rmyxlcmdaxmgpawmkf3fjhoz34ovg
```

**Comment by michaelkuhn** [comment]
Looks this cropped up before: https://groups.google.com/forum/#!msg/spack/XekK4Shxn3M/q8oo05I6AgAJ

My installation of HDF5 (with MPI support) does not contain a libhdf5_mpi, so it might be necessary to patch the build system to look for libhdf5.

**Comment by rajeeja** [comment]
Right. The does not have a solution. What you propose means a changes to visit spack package.

**Comment by steffen-AEI** [comment]
> This is a known bug in the concretizer that @tgamblin is currently working on fixing. In the meantime, the following workaround should work:
> 
> ```
> $ spack install visit ^vtk+python+qt ^python@:2.8
> ```

55 days before the end of Python2, there's no Python3 solution for this? *scratches head*

**Comment by sethrj** [comment]
@steffen-AEI Yeah, VisIt (and likely many other lab-created software packages out there) hasn't prioritized an upgrade to Python 3.

@rajeeja I reproduced your error with my mac toolchain (using Spack-installed HDF5 and OpenMPI):
```console
$ spack spec visit ^silo+mpi ^vtk+python+qt ^hdf5+mpi ^python@2.7.16
Input spec
--------------------------------
visit
    ^hdf5+mpi
    ^python@2.7.16
    ^silo+mpi
    ^vtk+python+qt

Concretized
--------------------------------
==> Warning: clang@11.0.0-apple cannot build optimized binaries for "skylake". Using best target possible: "x86_64"
visit@3.0.1%clang@11.0.0-apple~adios2 build_type=RelWithDebInfo +gui+hdf5+mpi patches=b1c10e8f99fd869cc488e7c17e135849c7f0d4ca4414d18a6760247188114031 +python+silo arch=darwin-mojave-x86_64
    ^cmake@3.15.2%clang@11.0.0-apple~doc+ncurses+openssl+ownlibs~qt arch=darwin-mojave-x86_64
    ^hdf5@1.10.5%clang@11.0.0-apple~cxx~debug~fortran+hl+mpi patches=b61e2f058964ad85be6ee5ecea10080bf79e73f83ff88d1fa4b602d00209da9c +pic+shared+szip~threadsafe arch=darwin-mojave-x86_64
        ^libszip@2.1.1%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^openmpi@3.1.4%clang@11.0.0-apple~cuda+cxx_exceptions fabrics=none ~java~legacylaunchers~memchecker~pmi schedulers=none ~sqlite3~thread_multiple+vt arch=darwin-mojave-x86_64
            ^hwloc@1.11.11%clang@11.0.0-apple~cairo~cuda~gl+libxml2~nvml~pci+shared arch=darwin-mojave-x86_64
                ^libxml2@2.9.9%clang@11.0.0-apple~python arch=darwin-mojave-x86_64
                    ^libiconv@1.16%clang@11.0.0-apple arch=darwin-mojave-x86_64
                    ^pkgconf@1.6.3%clang@11.0.0-apple arch=darwin-mojave-x86_64
                    ^xz@5.2.4%clang@11.0.0-apple arch=darwin-mojave-x86_64
                    ^zlib@1.2.11%clang@11.0.0-apple+optimize+pic+shared arch=darwin-mojave-x86_64
    ^python@2.7.16%clang@11.0.0-apple+bz2+ctypes+dbm+lzma~nis~optimizations patches=210df3f28cde02a8135b58cc4168e70ab91dbf9097359d05938f1e2843875e57 +pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib arch=darwin-mojave-x86_64
        ^bzip2@1.0.8%clang@11.0.0-apple+shared arch=darwin-mojave-x86_64
            ^diffutils@3.7%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^expat@2.2.9%clang@11.0.0-apple~libbsd arch=darwin-mojave-x86_64
        ^gdbm@1.18.1%clang@11.0.0-apple arch=darwin-mojave-x86_64
            ^readline@8.0%clang@11.0.0-apple arch=darwin-mojave-x86_64
                ^ncurses@6.1%clang@11.0.0-apple~symlinks~termlib arch=darwin-mojave-x86_64
        ^gettext@0.20.1%clang@11.0.0-apple+bzip2+curses+git~libunistring+libxml2+tar+xz arch=darwin-mojave-x86_64
            ^tar@1.32%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^libffi@3.2.1%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^openssl@1.1.1d%clang@11.0.0-apple+systemcerts arch=darwin-mojave-x86_64
            ^perl@5.30.0%clang@11.0.0-apple+cpanm+shared+threads arch=darwin-mojave-x86_64
        ^sqlite@3.30.1%clang@11.0.0-apple~column_metadata+fts~functions~rtree arch=darwin-mojave-x86_64
    ^qt@5.13.1%clang@11.0.0-apple~dbus~examples+framework freetype=none ~gtk~krellpatch+opengl patches=7f34d48d2faaa108dc3fcc47187af1ccd1d37ee0f931b42597b820f03a99864c,c52f72dac7fdff5a296467536cc9ea024d78f94b49903286395f53fd0eb66e5e ~phonon+shared~sql~ssl~tools~webkit arch=darwin-mojave-x86_64
        ^double-conversion@2.0.1%clang@11.0.0-apple build_type=RelWithDebInfo arch=darwin-mojave-x86_64
        ^glib@2.56.3%clang@11.0.0-apple~libmount patches=c325997b72a205ad1638bb3e3ba0e5b73e3d32ce63b2d0d3282f3e3a2ff4663c tracing=none arch=darwin-mojave-x86_64
            ^pcre@8.42%clang@11.0.0-apple~jit+multibyte+utf arch=darwin-mojave-x86_64
        ^harfbuzz@2.3.1%clang@11.0.0-apple arch=darwin-mojave-x86_64
            ^cairo@1.16.0%clang@11.0.0-apple~X~gobject~pdf arch=darwin-mojave-x86_64
                ^fontconfig@2.12.3%clang@11.0.0-apple arch=darwin-mojave-x86_64
                    ^font-util@1.3.2%clang@11.0.0-apple fonts=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,font-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-ttf,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-speedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiopic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 arch=darwin-mojave-x86_64
                        ^autoconf@2.69%clang@11.0.0-apple arch=darwin-mojave-x86_64
                        ^automake@1.16.1_1%clang@11.0.0-apple arch=darwin-mojave-x86_64
                        ^bdftopcf@1.0.5%clang@11.0.0-apple arch=darwin-mojave-x86_64
                            ^fontsproto@2.1.3%clang@11.0.0-apple arch=darwin-mojave-x86_64
                                ^util-macros@1.19.1%clang@11.0.0-apple arch=darwin-mojave-x86_64
                            ^libxfont@1.5.2%clang@11.0.0-apple arch=darwin-mojave-x86_64
                                ^freetype@2.10.1%clang@11.0.0-apple arch=darwin-mojave-x86_64
                                    ^libpng@1.6.37%clang@11.0.0-apple arch=darwin-mojave-x86_64
                                ^libfontenc@1.1.3%clang@11.0.0-apple arch=darwin-mojave-x86_64
                                    ^xproto@7.0.31%clang@11.0.0-apple arch=darwin-mojave-x86_64
                                ^xtrans@1.3.5%clang@11.0.0-apple arch=darwin-mojave-x86_64
                        ^mkfontdir@1.0.7%clang@11.0.0-apple arch=darwin-mojave-x86_64
                            ^mkfontscale@1.1.2%clang@11.0.0-apple arch=darwin-mojave-x86_64
                    ^gperf@3.0.4%clang@11.0.0-apple arch=darwin-mojave-x86_64
                ^pixman@0.38.0%clang@11.0.0-apple arch=darwin-mojave-x86_64
            ^icu4c@64.1%clang@11.0.0-apple cxxstd=11 arch=darwin-mojave-x86_64
        ^libjpeg-turbo@2.0.3%clang@11.0.0-apple arch=darwin-mojave-x86_64
            ^nasm@2.14.02%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^libmng@2.0.3%clang@11.0.0-apple arch=darwin-mojave-x86_64
            ^lcms@2.9%clang@11.0.0-apple arch=darwin-mojave-x86_64
                ^libtiff@4.0.10%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^opengl@2.1%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^pcre2@10.31%clang@11.0.0-apple+multibyte arch=darwin-mojave-x86_64
    ^qwt@6.1.3%clang@11.0.0-apple~designer patches=73df7272a11effc11546ab63aec4fc5ee25983d631c3fe5df024db3be8e83a1e arch=darwin-mojave-x86_64
    ^silo@4.10.2%clang@11.0.0-apple+fortran+mpi patches=7b5a1dc2a0e358e667088d77e7caa780967fa8ea60be89c44986605df9990abe +pic+shared~silex arch=darwin-mojave-x86_64
    ^vtk@8.1.2%clang@11.0.0-apple build_type=RelWithDebInfo ~ffmpeg~mpi+opengl2~osmesa+python+qt~xdmf arch=darwin-mojave-x86_64
        ^glew@2.0.0%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^jsoncpp@1.9.1%clang@11.0.0-apple build_type=RelWithDebInfo cxxstd=default arch=darwin-mojave-x86_64
        ^lz4@1.9.2%clang@11.0.0-apple arch=darwin-mojave-x86_64
        ^netcdf-c@4.7.2%clang@11.0.0-apple~dap~hdf4 maxdims=1024 maxvars=8192 +mpi~parallel-netcdf patches=56532470875b9a97f3cf2a7d9ed16ef1612df3265ee38880c109428322ff3a40,f52db13c61b9c19aafe03c2a865163b540e9f6dee36e3a5f808f05fac59f2030 +pic+shared arch=darwin-mojave-x86_64
            ^m4@1.4.18%clang@11.0.0-apple patches=3877ab548f88597ab2327a2230ee048d2d07ace1062efe81fc92e91b7f39cd00,c0a408fbffb7255fcc75e26bd8edab116fc81d216bfd18b473668b7739a4158e,fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c8 +sigsegv arch=darwin-mojave-x86_64
        ^netcdf-cxx@4.2%clang@11.0.0-apple+netcdf4 arch=darwin-mojave-x86_64
```
and got the error:
```
-- Looking for HDF5_MPI
CMake Error at CMake/SetUpThirdParty.cmake:214 (MESSAGE):
  Library hdf5_mpi not found in
  /rnsdhpc/code/spack/opt/spack2/clang/hdf5/tvz743p/lib
Call Stack (most recent call first):
  CMake/FindHDF5.cmake:76 (SET_UP_THIRD_PARTY)
  CMakeLists.txt:1289 (INCLUDE)
```

I'll see if I can create a patchfile to fix this. I'm having a bad sense of déjà vu... plus I'm sure I was able to install visit through spack with MPI before!

**Comment by steffen-AEI** [comment]
OK, it's going to stay im limbo then as I'm refusing to install python@:2.8 - there's enough trouble for me maintaining multiple versions of other packages, no need to duplicate python-related efforts. Thanks for clarifying.

**Comment by rajeeja** [comment]
@steffen-AEI Sigh! It is surprisingly hard to install visit on the new Ubuntu 19.04!

**Comment by rajeeja** [comment]
Thanks everyone!  and @adamjstewart  specially. 
I updated spack and was able to install visit.
Now, I get: 

```
jain@jain:~/spack$ 
==[develop]> alias python='python2.7';PYTHONPATH=/usr/lib/python2.7/ /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/visit-3.0.1-srfp77b3n2j2gwomqyeqfjatb7p73i3w/bin/visit
Fatal Python error: initfsencoding: Unable to get the locale encoding
  File "/usr/lib/python2.7/encodings/__init__.py", line 123
    raise CodecRegistryError,\
                            ^
SyntaxError: invalid syntax

Current thread 0x00007fe97932fb80 (most recent call first):
Aborted (core dumped)
```

**Comment by rajeeja** [comment]
How do I launch visit? I have python 3.7 as my default. My current line is:
```
alias python='python2.7';
PYTHONPATH=/usr/lib/python2.7/ /home/jain/spack/opt/spack/linux-ubuntu19.04-sandybridge/gcc-8.3.0/visit-3.0.1-srfp77b3n2j2gwomqyeqfjatb7p73i3w/bin/visit
```

**Comment by sethrj** [comment]
I'd guess that `alias python` isn't good enough (since that won't affect calls from forked processes); try `export PATH=$(spack location -i python@:3.0)/bin:$PATH`

**Comment by adamjstewart** [comment]
PYTHONPATH should use colons to separate paths, not spaces. I've never used visit before, but I would be surprised if it's Python libraries were stored in `prefix.bin`.

**Comment by sethrj** [comment]
@adamjstewart I think he's using the `KEY=VALUE EXE` syntax to set the python path to the system value and launch visit with that. Not recommended of course.

**Comment by rajeeja** [comment]
Assume visit in my PATH and I have python3 system installed as default. Let me know how to specify python correctly?
```
which python
/usr/bin/python
jain@jain:~$ 
==> python --version
Python 3.7.3
```

```
==>  spack find -p python
3 installed packages
-- linux-ubuntu18.04-x86_64 / gcc@7.3.0 -------------------------
python@2.7.12  /usr

-- linux-ubuntu19.04-sandybridge / gcc@8.3.0 --------------------
**python@2.7.6   /usr**
python@2.7.12  /usr
```
visit depends on 2.7.6.

**Comment by sethrj** [comment]
I would guess that visit will only work once `sh python` evaluates to a 2.x version. It looks like you've directed Spack to use the system installations of Python, which probably breaks a number of assumptions spack makes. It's *rarely* if ever a good idea to make external spack packages that point at system installations. I recommend removing all the `buildable=False` and external paths from your `packages.yaml` file, and rebuilding your toolchain.

**Comment by Rihas** [comment]
hi There this works for me.
spack install visit ^python+shared ^glib@2.56.3 ^py-setuptools@44.1.0

**Comment by alalazo** [comment]
Duplicate of #1781

**Comment by alalazo** [comment]
The concretization issue is a duplicate of #1781 and will be solved by #19501. I'll leave this issue open since the discussion drifted to build issues, but feel free to close it if everything was solved.

**Comment by iulian787** [comment]
Rajeev,
Where you able to build visit + mpi + hdf5 with spack? the library hdf5_mpi was renamed by Mark Miller, to differentiate it from hdf5 built serially (when you build in parallel, visit keeps both versions, serial and parallel of many libraries); 
It was needed for moab :)

**Comment by rajeeja** [comment]
Yes, it finally worked at the time.

_________________________________________
Rajeev
